### PR TITLE
CIV-3023: PIP Validating the Pin 

### PR DESCRIPTION
**Pull request by kenneth-hmcts**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/CIV-3023

### Change description ###
This PR contains changes for checking claim reference and validate the Pin part for the LRvLiP Pin in Post

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/civil/service/pininpost/DefendantPinToPostLRspecService.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import static uk.gov.hmcts.reform.civil.callback.CaseEvent.UPDATE_CASE_DATA;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class DefendantPinToPostLRspecService {
+
+    private final CoreCaseDataService coreCaseDataService;
+
+    public void checkPinValid(CaseData caseData, String pin) {
```
Comment (TatsyMaslova) [inline src/main/java/uk/gov/hmcts/reform/civil/service/pininpost/DefendantPinToPostLRspecService.java:27]: I haven't noticed it before but perhaps we should rename method to validatePin?

**Review by TatsyMaslova — APPROVED** [review]

**Review by TatsyMaslova — APPROVED** [review]

**Review by hmcts-version1-domini — APPROVED** [review]

**Review by hmcts-version1-domini — DISMISSED** [review]

**Review by TatsyMaslova — DISMISSED** [review]

**Review by TatsyMaslova — DISMISSED** [review]

File: src/integrationTest/java/uk/gov/hmcts/reform/civil/controllers/cases/CaseAssignmentControllerTest.java [hunk 2]
```diff
@@ -52,18 +52,18 @@ void givenIncorrectReference_whenValidateCaseAndPin_shouldReturnUnauthorised() {
     void givenIncorrectPin_whenValidateCaseAndPin_shouldReturnUnauthorised() {
         givenCaseIsFound();
         doThrow(new PinNotMatchException()).when(defendantPinToPostLRspecService).validatePin(
-            any(CaseData.class),
+            any(CaseDetails.class),
             anyString()
         );
 
         doPost("", "123", VALIDATE_PIN_URL, "123")
```
Comment (TatsyMaslova) [inline src/integrationTest/java/uk/gov/hmcts/reform/civil/controllers/cases/CaseAssignmentControllerTest.java:59]: in this case we should test that it returns bad request

Comment (kenneth-hmcts) [reply src/integrationTest/java/uk/gov/hmcts/reform/civil/controllers/cases/CaseAssignmentControllerTest.java:59]: Changes have been made in the CaseAssignmentController

Comment (TatsyMaslova) [reply src/integrationTest/java/uk/gov/hmcts/reform/civil/controllers/cases/CaseAssignmentControllerTest.java:59]: i understand but we need to make changes in the test. So we need to change the title and what status we expect. In the title of the test we are expecting it to return unauthorised. but we should expect it to return 400

Comment (TatsyMaslova) [reply src/integrationTest/java/uk/gov/hmcts/reform/civil/controllers/cases/CaseAssignmentControllerTest.java:59]: cool can see your changes in the correct test but need to change the titele of the test _shouldReturnBadRequest

**Review by TatsyMaslova — DISMISSED** [review]

**Review by hmcts-version1-domini — APPROVED** [review]

**Review by TatsyMaslova — APPROVED** [review]

**Pull request merged**